### PR TITLE
Support embargoed Dandisets

### LINKED ISSUE #8: Support embargoed dandisets
ATM we are completely skipping them. In the light of integration with github for issue tracking etc would be desired/needed.  In general it should not require much difference in behavior I think, but @jwodder please correct me if I am wrong. I see the need to:

- Created github repos would be created private
- For all assets populate only API URLs
  - @yarikoptic would need to look into how we should handle authentication. Likely the simplest way would be to provide (within DataLad) a "provider" configuration to provide the API Token in the request. 
  - I do not think we should populate direct URLs to private bucket ATM at all, in particular since embargo is being [redesigned](https://github.com/dandi/dandi-archive/pull/1772), but we could for "public assets" in embargoed datasets.
- If dandiset becomes unembargoed 
  - github repository should be made public if private
  - we should populate direct http URLs to S3 for all blobs  (worth adding a service command for that: given a dandiset id go through all assets and ensure presence of direct s3 http url)

Some related thoughts
- since dropbox backup is "private" anyways, no need to anyhow separate it out for embargoed content

@AlmightyYakob , a few questions:

- in the API ATM can we know if blob is embargoed or not -- from URL?
- after [embargo redesign](https://github.com/dandi/dandi-archive/pull/1772) how would we know that?
- do you have an example blob or just a URL to some key configured like [designed](https://github.com/dandi/dandi-archive/pull/1772/files#diff-f3d72d358571f346f0a3e26e3bf309127d6af6aac1e39f60e291b947d86eeb19R29) so that access would be denied -- I wonder if response would provide detail on how to authenticate to gain access. I.e. would it be 403 or [401](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401) with useful WWW-Authenticate

### PR DESCRIPTION
**Pull request by jwodder**
Closes #8.

To do:

- [x] Decide how to handle assets missing SHA256 digests when registering S3 URLs (https://github.com/dandi/backups2datalad/issues/8#issuecomment-1874502017)
- [x] When a Dandiset is unembargoed, make the GitHub repository public
- [x] Add a command for just registering S3 URLs
- [x] Add tests

### CODE REVIEW

**Comment by jwodder** [comment]
@yarikoptic Ping.

File: README.md [hunk 1]
```diff
@@ -20,6 +20,10 @@ Before running `backups2datalad`, the following setup must be performed:
   version 10.20230126 is required, though you should endeavor to obtain the
   latest version.
 
+- An API token needs to be obtained for the DANDI instance that is being
+  mirrored.  When invoking `backups2datalad`, the environment variable
+  `DANDI_API_KEY` must be set to the token.
```
Comment (yarikoptic) [inline README.md:25]: do you recall if we have already any cron job on drogon which needs and uses it so we could centralize secrets specification?  couldn't recall any from a quick look.

Comment (yarikoptic) [reply README.md:25]: I guess we would need to add `source`ing the file with secrets at https://github.com/dandi/dandisets/blob/draft/tools/backups2datalad-cron-common  or do you see better place?

Comment (jwodder) [reply README.md:25]: I don't believe we have any other cronjobs that access the Archive, let alone with an API key.

Comment (yarikoptic) [reply README.md:25]: 1ok, done changes to the script, let's see ;-)  Thank you!

**Pull request merged**